pkg/xds/generator/ingress_generator.go

```diff
@@ -129,6 +157,49 @@ func (_ IngressGenerator) destinations(
 		}
 	}
 
+	if len(ingressProxy.PolicyResources[meshhttproute_api.MeshHTTPRouteType].GetItems()) > 0 {
+		// We need to add a destination to route any service to any instance of
+		// that service
+		matchAllTags := tags.Tags{
+			mesh_proto.ServiceTag: mesh_proto.MatchAllTag,
+		}
+		matchAllDestinations := destinations[mesh_proto.MatchAllTag]
+		foundAllServicesDestination := slices.IndexFunc(matchAllDestinations, func(tagsElem tags.Tags) bool {
+			return reflect.DeepEqual(tagsElem, matchAllTags)
+		}) > -1
+		if !foundAllServicesDestination {
```

Comment:
definitely not blocking, but this took me some time to get, maybe it would be better to put the `> -1` to the if statement?

Reply:
Ah nice, finally a `slices.ContainsFunc` added in December.